File: iOS Example/iOS OC Example/VideoViewController.m
//
//  VideoViewController.m
//  iOS OC Example
//
//  Created by 公司 on 2018/6/13.
//

#import "VideoViewController.h"

@import AdTiming;
@interface VideoViewController () <ADTVideoAdDelegate>

@end

@implementation VideoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)load:(id)sender {
    [ADTVideoAd sharedInstance].delegate = self;
    [[ADTVideoAd sharedInstance]loadWithPlacmentID:@"113"];
}

- (IBAction)show:(id)sender {
    [[ADTVideoAd sharedInstance] isReady:@"113"];
    [[ADTVideoAd sharedInstance]show:@"113"];
}

- (void)ADTVideoAdDidload:(NSString*)placementID{
    NSLog(@"广告加载成功");
}

- (void)ADTVideoAdDidFailToLoad:(NSError*)error{
    NSLog(@"广告加载成功");
}

- (void)ADTVideoAdDidStart{
    NSLog(@"广告开始播放");
}

- (void)ADTVideoAdDidClick{
    NSLog(@"点击广告");
}

- (void)ADTVideoAdDidFinish:(ADTVideoAdFinishState)state{
    if(state == ADTVideoAdFinishStateSkipped){
        NSLog(@"广告播放完成(跳过)");
    } else if (state == ADTVideoAdFinishStateCompleted){
        NSLog(@"广告播放完成(完整)");
    }
}

@end

File: iOS Example/iOS Example/VideoViewController.swift
//
//  VideoViewController.swift
//  iOS Example
//
//  Created by 公司 on 2018/6/13.
//

import UIKit

class VideoViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func load(_ sender: Any) {
        ADTVideoAd.sharedInstance().delegate = self
        ADTVideoAd.sharedInstance().load(withPlacmentID: "113")
    }
    
    @IBAction func show(_ sender: Any) {
        ADTVideoAd.sharedInstance().isReady("113")
        ADTVideoAd.sharedInstance().show("113")
    }
    
}

extension VideoViewController: ADTVideoAdDelegate {
    func adtVideoAdDidload(_ placementID: String) {
        print("videoAdDidLoad")
    }
    
    func adtVideoAdDidFail(toLoad error: Error) {
        print("videoAdShowFail")
    }
    
    func adtVideoAdDidStart() {
        print("VideoAdStartPlay")
    }
    
    func adtVideoAdDidClick() {
        print("videoAdDidClick")
    }
    
    func adtVideoAdDidFinish(_ state: ADTVideoAdFinishState) {
        if state == .skipped {
            print("VideoAdSkipped")
        } else if state == .completed {
            print("VideoAdCompleted")
        }
    }
}
